flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaWriterITCase.java

```diff
@@ -32,40 +40,44 @@
 
 import org.apache.kafka.clients.producer.ProducerRecord;
 import org.apache.kafka.common.serialization.ByteArraySerializer;
-import org.junit.Before;
-import org.junit.ClassRule;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testcontainers.containers.KafkaContainer;
 import org.testcontainers.containers.Network;
 import org.testcontainers.containers.output.Slf4jLogConsumer;
 import org.testcontainers.utility.DockerImageName;
 
+import java.io.IOException;
+import java.nio.ByteBuffer;
 import java.time.Duration;
-import java.util.List;
+import java.util.Map;
 import java.util.Properties;
+import java.util.stream.IntStream;
 
+import static org.hamcrest.Matchers.greaterThan;
 import static org.junit.Assert.assertThrows;
 
 /** Tests for the standalone KafkaWriter. */
-@RunWith(Parameterized.class)
 public class KafkaWriterITCase {
 
     private static final Logger LOG = LoggerFactory.getLogger(KafkaWriterITCase.class);
     private static final Slf4jLogConsumer LOG_CONSUMER = new Slf4jLogConsumer(LOG);
     private static final String INTER_CONTAINER_KAFKA_ALIAS = "kafka";
     private static final Network NETWORK = Network.newNetwork();
     private static final String KAFKA_METRIC_WITH_GROUP_NAME = "KafkaProducer.incoming-byte-total";
-
-    private final DeliveryGuarantee guarantee;
+    private static final SinkWriter.Context SINK_WRITER_CONTEXT = new DummySinkWriterContext();
 
     private MetricListener metricListener;
 
-    @ClassRule
-    public static final KafkaContainer KAFKA_CONTAINER =
+    private static final KafkaContainer KAFKA_CONTAINER =
```

Comment:
Add `@Nested` test? Then you don't need to spawn container when running a test in IDE that doesn't need it.

Reply:
I think all the tests need the KafkaContainer or what are you referring to?

Reply:
Okay sorry didn't see that. Then of course, leave as is.